python/ray/serve/tests/test_http_state.py

```diff
@@ -570,6 +570,10 @@ async def check_health(self):
     def _update_until_num_health_checks_received(
         state: HTTPProxyState, num_health_checks: int
     ):
+        # state.update should not be called in the terminal states.
+        if state.status in {ProxyStatus.UNHEALTHY, ProxyStatus.DRAINED}:
+            return True
```

Comment:
is this guaranteed in the production codepath already? probably better to handle this case elegantly

Reply:
Or I guess we can just add an early return in the update so it becomes a no-op in those terminal states

Reply:
that sounds better -- be defensive against ourselves

Reply:
should also remove this early return then right?